app/app.go

```diff
@@ -795,26 +797,36 @@ func (app *PylonsApp) RegisterTendermintService(clientCtx client.Context) {
 	)
 }
 
-// RegisterUpgradeHandlers returns upgrade handlers
-func (app *PylonsApp) RegisterUpgradeHandlers(cfg module.Configurator) {
-	app.UpgradeKeeper.SetUpgradeHandler(upgradev46.UpgradeName, upgradev46.CreateUpgradeHandler(app.mm, app.configurator, &app.StakingKeeper, app.keys[pylonsmoduletypes.StoreKey], app.appCodec))
+// setupUpgradeHandlers setup upgrade handlers
+func (app *PylonsApp) setupUpgradeHandlers() {
+	// v3 upgrade handler
+	app.UpgradeKeeper.SetUpgradeHandler(
```

Comment:
Why we `SetUpgradeHandler` of old update plan here ?

Reply:
It's really no need for that but I think instead of deleting the old update plan like above we just need to add the new update plan when upgrading, the new guys can easily read the versions upgrade and it does not take a lot of costs when upgrade, I had referenced this from Osmosis and Evmos

Reply:
Osmosis:
https://github.com/osmosis-labs/osmosis/blob/da175c5d8558655ace3702718030c85a0eaf2bc7/app/app.go#L420
Evmos:
https://github.com/evmos/evmos/blob/af003d052033912ac8723941fefd6321422ea41d/app/app.go#L1031